modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4TcpChannel.java

```diff
@@ -142,11 +145,20 @@ public InetSocketAddress getRemoteAddress() {
     }
 
     @Override
-    public void sendMessage(BytesReference reference, ActionListener<Void> listener) {
-        channel.writeAndFlush(Netty4Utils.toByteBuf(reference), addPromise(listener, channel));
-
-        if (channel.eventLoop().isShutdown()) {
-            listener.onFailure(new TransportException("Cannot send message, event loop is shutting down."));
+    public void sendMessage(OutboundHandler.SendContext sendContext) {
+        try {
+            channel.eventLoop().execute(() -> {
```

Comment:
Can we do this in `Netty4MessageChannelHandler#write`? So that any uncaught exceptions going through the normal channel exception handling?

Reply:
Done in 6a85d40, that's much better indeed thanks :)